src/syscheckd/db/fim_db_registries.c

```diff
@@ -558,7 +520,7 @@ int fim_db_insert_registry(fdb_t *fim_sql, fim_entry *new) {
     int res_key = 0;
 
     res_key = fim_db_insert_registry_key(fim_sql, new->registry_entry.key, new->registry_entry.key->id);
-    fim_db_get_registry_key_rowid(fim_sql, new->registry_entry.key->path, &new->registry_entry.key->id);
+    fim_db_get_registry_key_rowid(fim_sql, new->registry_entry.key->path, new->registry_entry.key->arch, &new->registry_entry.key->id);
```

Comment:
Please, don't exceed 120 characters per line.
```suggestion
    fim_db_get_registry_key_rowid(fim_sql, new->registry_entry.key->path, new->registry_entry.key->arch,
                                  &new->registry_entry.key->id);
```